File: medplum-java/medplum-model/src/main/java/com/fhir/CarePlan_Activity.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCarePlan_Activity.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface CarePlan_Activity {
  Optional<String> id();

  Optional<List<CodeableConcept>> outcomeCodeableConcept();

  Optional<List<Annotation>> progress();

  Optional<Reference> reference();

  Optional<CarePlan_Detail> detail();

  Optional<List<Reference>> outcomeReference();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  static ImmutableCarePlan_Activity.Builder builder() {
    return ImmutableCarePlan_Activity.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/CarePlan_Activity.ts
import { Annotation } from "./Annotation";
import { CarePlan_Detail } from "./CarePlan_Detail";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Reference } from "./Reference";

export interface CarePlan_Activity {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    outcomeCodeableConcept: CodeableConcept[] | null | undefined;
    outcomeReference: Reference[] | null | undefined;
    progress: Annotation[] | null | undefined;
    reference: Reference | null | undefined;
    detail: CarePlan_Detail | null | undefined;
}
